secure/storage/src/tests/vault.rs

```diff
@@ -181,6 +195,90 @@ fn test_vault_key_value_policies() {
     assert_eq!(storage.get::<u64>(ROOT).unwrap().value, 2);
     assert_eq!(storage.get::<u64>(PARTIAL).unwrap().value, 7);
     assert_eq!(storage.get::<u64>(FULL).unwrap().value, 12);
+
+    // Verify a reader with another namespace has no permission for the operations
+    let mut reader_with_namespace = VaultStorage::new(
+        dev::test_host(),
+        reader_token,
+        Some(VAULT_NAMESPACE_1.into()),
+        None,
+        ttl,
+        false,
+    );
+    assert_eq!(
```

Comment:
Perhaps we could just check one of these keys to make sure different namespaces really aren't accessible? Checking of all them seems a little excessive to me.